src/clj/puppetlabs/puppetserver/cli/subcommand.clj

```diff
@@ -30,16 +30,25 @@
    The given `run-fn` is then called with the configuration map and the command
    line arguments specific to the subcommand.
 
-   The `run-fn` must accept a configuration map and argument sequence."
+   The `run-fn` must accept a configuration map and argument sequence.
+   The configuration map will look like:
+    {
+      :config       {<the fully parsed & merged configuration map>}
+      :config-path  '/opt/puppetlabs/server/app/conf.d/'
+      :config-files [<File foo.conf>, <File bar.conf>, <File qux.conf>]
+    }
```

Comment:
I struggled to determine how best to support the data hand-off between this general function and the specific subcommand "run" functions.  
The problem was that it only provided the final, merged map of configuration settings to the subcommand, but I needed the actual file names so I could write to them.  For read-only actions, I could reuse the final, merged map though, so I kind of wanted access to both.

That's how I ended up with generalizing the existing "config" parameter (that's passed down to subcommand run functions) from just the final, merged map to instead encompassing all the configuration-related info (which is why I put the :config-path in there even though I wasn't using it in this subcommand).

I also played with the idea of having multiple `cli/run` functions that subcommand could call, but in the end I went with the option that required the fewest code changes.

Suggestions very much welcome for this part.